conanfile.py

```diff
@@ -617,7 +615,10 @@ def _configure_autotools(self):
         args.append("--with-libz={}".format("yes" if self.options.with_zlib else "no"))
         if self._has_with_libdeflate_option:
             args.append("--with-libdeflate={}".format("yes" if self.options.with_libdeflate else "no"))
-        args.append("--with-libiconv-prefix={}".format(tools.unix_path(self.deps_cpp_info["libiconv"].rootpath)))
+        if self.options.with_libiconv:
+            args.append("--with-libiconv-prefix={}".format(tools.unix_path(self.deps_cpp_info["libiconv"].rootpath)))
+        else:
+            args.append("--without-libiconv-prefix")
```

Comment:
```suggestion
        args.append("--with-libiconv-prefix={}".format(tools.unix_path(self.deps_cpp_info["libiconv"].rootpath) if self.options.with_libiconv  else "no"))
```

can be simplified in one line

Reply:
I think you actually need to be verbose in disabling it

Reply:
https://github.com/OSGeo/gdal/blob/master/gdal/m4/iconv.m4#L35-L40

Reply:
`--with-libiconv-prefix=no` should be the same than `--without-libiconv-prefix`